src/consensus/dkg.rs

```diff
@@ -105,13 +108,10 @@ impl DkgVoter {
         keypair: &Keypair,
         dkg_key: DkgKey,
         elders_info: EldersInfo,
-        key_index: u64,
     ) -> Vec<DkgCommand> {
-        if let Some(session) = self.sessions.get(&dkg_key) {
-            if !session.complete && session.key_index >= key_index {
-                trace!("DKG for {} already in progress", elders_info);
-                return vec![];
-            }
+        if self.sessions.contains_key(&dkg_key) {
```

Comment:
`DkgKey` now contains the generation number to prevent messages from different DKGs that have the same participants but different generation from being mixed up. That was causing a DKG to seemingly complete successfully, but the generated secret shares were invalid and were producing invalid signature shares. And to answer your question: we now only prevent starting an already started DKG. However, when a new DKG is started, we check the other ongoing DKGs and remove the ones whose generation is strictly lower, because they are no longer useful.

Reply:
>  when a new DKG is started, we check the other ongoing DKGs and remove the ones whose generation is strictly lower, because they are no longer useful.

just to confirm that when take fork into account, say there is chain `0,1,2` first, then inserting `0->3`, 
will then the DKG on 2 got `cleaned out` due to have a lower generation than the DKG on 3 ?